### PR TITLE
Improve error message if serverLayer is reused.

### LINKED ISSUE #17129: http server blueprint is reused but cannot be
The server blueprint currently isn't reusable but it actually is reused in `HttpExt.bind`. Because reusage isn't prevented currently this currently introduces hard to detect errors.

The short-term solution should be not to reuse the HTTP server blueprint.

In the long-term it should be attempted to make the server blueprint reusable. Currently this isn't possible because it needs to access the ActorRef of the token source for Expect 100 support but there's no way to access it currently (can hopefully be fixed when #16168 is implemented). So, the token source writes the ActorRef into a mutable volatile var that is shared over all connections (!). Currently, the cell is overwritten which each new materialization of the blueprint, i.e. Expect 100 support only works on the latest connection and can be disrupted by older connections also trying to use Expect 100 support.

Also, for similar reasons initial websocket support won't be reusable for now.

### PR DESCRIPTION
**Pull request by jrudolph**
Fixes #17129 for now.

Based on @sirthias' #17283.

### CODE REVIEW

**Comment by jrudolph** [comment]
Added a commit which fixes a small issue in the Websocket implementation.

File: akka-stream/src/main/scala/akka/stream/impl/CompletedPublishers.scala [hunk 1]
```diff
@@ -53,7 +53,7 @@ private[akka] case object CancelledSubscription extends Subscription {
 /**
  * INTERNAL API
  */
-private[akka] case object RejectAdditionalSubscibers extends Publisher[Nothing] {
+private[akka] case object RejectAdditionalSubscribers extends Publisher[Nothing] {
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/CompletedPublishers.scala]: thanks

**Comment by patriknw** [comment]
LGTM

**Comment by jrudolph** [comment]
Failed in `akka.stream.io.TlsSpec`.

**Comment by jrudolph** [comment]
Rebased.

**Pull request merged**